lib/Backend/Lower.cpp

```diff
@@ -2896,7 +2896,23 @@ Lowerer::LowerRange(IR::Instr *instrStart, IR::Instr *instrEnd, bool defaultDoFa
         case Js::OpCode::StatementBoundary:
             // This instruction is merely to help convey source info through the IR
             // and eventually generate the nativeOffset maps.
-            break;
+#if DBG_DUMP && DBG
+            // If we have a JITStatementBreakpoint, then we should break on this statement
```

Comment:
This functionality is very useful especially to debug some large JIT'd code. I am wondering could we design it as generating `__debugbreak` for some special source statement, like `debugger;`? It is more easy for revising the location of bp. But the a drawback is it makes break unmodified JS code hard (like for some live website).